packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1962,6 +1969,27 @@ export class ContainerRuntime
 		this.removeAllListeners();
 	}
 
+	/**
+	 * Utility function to check if any blobs under a snapshot tree is missing and if so, then set
+	 * omitted flag to true for that snapshot tree.
+	 * @param snapshotTree - snapshotTree to be evaluated for missing blobs.
+	 * @param blobContents - blobContents of the snapshot.
+	 */
+	private evaluateSnapshotForOmittedBlobContents(
+		snapshotTree: ISnapshotTree,
+		blobContents: Map<string, ArrayBuffer>,
+	) {
+		for (const [_, id] of Object.entries(snapshotTree.blobs)) {
+			if (!blobContents.has(id)) {
+				snapshotTree.omitted = true;
```

Comment:
Instead of setting a value on each tree, why not return it? I.e. the function answers the question - does tree have all the blobs or not. And data store can use it to make its decisions on what to do.